File: lcp/2.py

def fraction(cont):
    res=[cont[len(cont)-1],1]
    for i in range(len(cont)-2,-1,-1):
        temp=res[1]
        res[1] = res[0]
        res[0]=res[0]*cont[i]+temp
    return res


if __name__=="__main__":
    cont=[i for i in range(1,11)]
    res=fraction(cont)
    print(res)

File: lcp/2.go
package main

import "fmt"

func fraction(cont []int) []int {
	n := len(cont)
	res := []int{cont[n-1], 1}
	for i := n - 2; i >= 0; i-- {
		temp := res[1]
		res[1] = res[0]
		res[0] = cont[i]*res[0] + temp
		fmt.Println(res)

	}
	return res
}
func main() {
	cont := []int{3, 2, 0, 2}
	res := fraction(cont)
	fmt.Println(res)
}
